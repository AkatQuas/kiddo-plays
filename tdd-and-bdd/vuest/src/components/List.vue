<template>
  <div>
    <ul>
      <li v-for="item in items" :key="item">{{ item }}</li>
    </ul>
    <p class="message">{{message}}</p>
  </div>
</template>

<script>
export default {
  name: "List",
  props: {
    items: {
      type: Array,
      required: true
    },
    message: {
      type: String,
      default: "default message"
    }
  }
};
</script>

