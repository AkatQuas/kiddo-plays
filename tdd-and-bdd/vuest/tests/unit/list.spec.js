import { shallowMount } from '@vue/test-utils';
import List from '@/components/List.vue';

describe('List.vue', () => {
  const factory = (props) => shallowMount(List, {
    propsData: props,
  });

  it('renders li for each item in props.item', () => {
    const items = ['1', '2'];
    const wrapper = factory({ items });
    expect(wrapper.findAll('li')).toHaveLength(items.length);
    expect(wrapper.findAll('li').length).toBe(items.length);
  });

  it('matches snapshot', () => {
    const items = ['item 1', 'item 2'];
    const wrapper = factory({ items });
    expect(wrapper.html()).toMatchSnapshot();

    expect(wrapper.find('.message').text()).toBe('default message');
  });

  it('render correct message', () => {
    const items = ['item 1', 'item 2'];
    const message = 'qaqaqa';
    const wrapper = factory({ items, message });

    expect(wrapper.find('.message').text()).toBe(message);
  });

  it('render correct message', () => {
    const items = ['item 1', 'item 2'];
    const wrapper = factory({ items  });

    expect(wrapper.find('.message').text()).toBe('default message');

    const message = '99999';
    wrapper.setProps({
      message,
    });
    expect(wrapper.find('.message').text()).toBe(message);

  });

})