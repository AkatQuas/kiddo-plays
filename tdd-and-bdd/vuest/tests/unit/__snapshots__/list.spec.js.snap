// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List.vue matches snapshot 1`] = `
<div>
  <ul>
    <li>item 1</li>
    <li>item 2</li>
  </ul>
  <p class="message">default message</p>
</div>
`;
